### DISCUSSION TITLE
Suggestions: new ideas for events and more

### DISCUSSION THREAD

**Discussion by eugenikus8** [Keriew/augustus#1120] [Ideas]
Tweak:
- [x] Trade ships leave the city through the river exit point (only if closer than river entry)
- [ ] Depots indestructible and non-flammable (fire_proof = 1)
- [x] Move "Roadblock" to the road construction menu
- [x] Move "Obelisk",  "Marble Column" to statues subcategory
- [ ] Move all "Ponds" to statues subcategory
- [ ] Move Monuments to bottom of the list
- [ ] Move "Engineering post" and "Prefecture" to top of lists.
- [x] Add City sounds for cart depo, prefecture, eng.post, acrhitect guild

<br>

Editor:
- [ ] Earthquake tiles can be cleared. Doing so requires stone, sand (or clay), and a workers' camp to deliver the resources
- [ ] Random earthquakes can be turned into a recurring rare event
- [ ] Add several markers for random earthquakes, along with the ability to clear them
- [ ] Ability to place bridges in editor
- [x] Add an option to display grid offset numbers
- [x] Custom earthquake (draw it like roads)
- [ ] Request: full random, random food, random raw material, random goods. Random resources are selected only from those available in the current mission.
- [x] New action: Centering camera on a map area
- [x] Color highlight the Grid Offset for the event. Also show event id for this grid offset
- [ ] Random events: add new types of mines that can collapse. Treat the sand quarry the same as the clay pit, and the gold mine the same as the iron mine. Also add collapse events for marble and stone quarries.
- [ ] Chance of triggering events
- [x] Name of events
- [ ] Permutation of Events, Conditions, and Actions
- [x] Copy/paste Events, Conditions, and Actions
- [x] 'or' for multiple conditions
- [ ] 'or' for multiple actions
- [x] New condition: Comparation check of sales/buys of goods over year
- [x] New action: Installing/removing objects (stones, water)
- [x] New action: Change building prices
- [x] New action: Change production speed
- [ ] New action: Fires
- [ ] New action: Collapsing
- [ ] "Herd point" 5-8
- [x] New action: Victory or Loss

<br>

Feature:
- [ ] Allow markets to receive goods based on the assigned priority, similar to the labor priority system.
- [x] Display statistics for the year before last in the financial advisor window (implemented in the "Trade Ledger")
- [ ] "Resource carts only deliver to nearby workshops," similar to how "Farms and wharves deliver only to nearby granaries"
- [ ] Tooltips for menu options
- [ ] Defensive structure 1x1, passable for enemies. Reduces movement speed by half
- [x] Show current HP (towers, gatehouse) in building info window
- [ ] Capturing retreating enemies and turning them into slaves, with a random amount of money added to the city treasury for each one
- [ ] New level of palace (5x5). Need: silk (import only?) or luxury goods (made of gold?), 4 types of food, all temples, all entertaiments, possibly access to the Senate
- [ ] Invasions by ships
- [x] Granary || Warehouse: Allow/Disallow access for the native trader
- [x] Warehouses: orders can now have a minimum of 4 units per item
- [ ] Water animation in the aqueduct
- [x] Displaying desirability area for "Oracle" and "Lararium", as well as for all decoratives structures
- [x] New sound for moving legions
- [x] New overlay: enemy (everyone who knows how to fight: soldiers, enemies, prefects, lions, gladiators, etc.)
- [ ] Add Narrow stone gate (similar to Palisade gate, but with the strength of a Wall)

**Comment by zzzzzz1002003** [comment]
ide like a new building that by itself does distribution..   (like a cart depo) you chose what resource you need to distribute to x buildings and it doesn't grab 1 cart load at a time like the warehouse guys do. but instead grabs 4-6 and distributes them where needed in its area. so in your production areas 1 extra building is needed but does a much more efficient job of grabbing and distributing goods.  im thinking a 2x2 or if to be annoying to ppl 3x3.

**Comment by turgonpesce** [comment]
Hello there, i compiled a list some time ago with the editor changes i would suggest to add. I'll write there so you can consider to add on to your list, considering i don't really know what can be realistically be done and what can't.

New conditions:
- population count in area (all, patricians, plebs, slums)
- service coverage - example: schools coverage Greather than 'average'; barbers coverage less than or equal 'excelent'
- % city population taxed
- housing dwellers. - example: small houses dwellers greather than '1000'; large insulae dwellers 'less than or equal 500')
- check monument stage. - EXAMPLE: 'lighthouse' stage 0 (zero meaning there is no foundation yet); Grand temple to venus stage '2' (selecting the monument from a list similar to the disallow building feature)
- age population count - EXAMPLE: "age Greater than" 20 'all population count greather than or equal' 4000; 'age less than' 15 'patricians count less than' 200)

New Actions:
- adjust sentiment
- change governor rank
- force mothball/unmothball buildings using the same list of allow/disallow building
- change production of specific good/building. - EX: fruits +/-20% for x months (all fruit farms produce 20% faster); schools +/-50% for x months (schoolars spawn 50% more frequently)
- lock/unlock custom variable for 'x' months (so specific variables cannot change for a while even if there is a repeating action running)
- start landslides/sea storm for specific trade route
- stop/start immigration

Editor:
- set different entry points for traders
- set specific coordinates target (or area) for invaders
- set prices/quota for specific trade route, with condition checking prices and actions changing them

**Comment by BCfplox** [comment]
Hi, suggestion for more conditions and actions like:

Conditions:

- Number of soldiers per legion.
- Morale level of each legion.
- Months of stored food for legions.
- Education, entertainment, health (coverage of the city by each type of construction(as sub menu) ).
- Number of pensioners, adults and children.
- Number of assigned manpower (needed and available).
- All financial variables (as submenu: taxes collected, business income, net flow, etc.):
- Number of each resource in residence used.
- Total number of trade per good in a (purchase or sale in the year / month ).

Very useful to compare the value of a condition parameter from another condition parameter or custom variable (Example: Prosperity equals Tax Rate).

Actions:

This action is very useful: assign to a custom variable a condition variable (and/or apply calculations) (Example: Change value of custom variable by adding Tax rate to it (also if it can be negative value ) .

I hope the last two paragraphs mentioned above take precedence as it makes the events dynamic and automated instead of almost everything depending on the value assigned manually by the map designer.

**Comment by mpdmtr** [comment]
Events:  
- Add puddles to terrain when it's raining, add random flood event during heavy rain (the heaviness depends from terrain type and can be regulated by map creator). The puddles and flood disappear automatically after some time of sunny weather (a week or 10 days).  

The degree of flood is also can be decreased by building drainage canals. Player can build walkways over them. 

- Replace the terrain with snowy one when it's snowing 
- Add small sand dunes after sandstorm (amount can be regulated by the creator). The can be cleansized by demolishion tool.

**Comment by Filu2259** [comment]
Hi, here are some suggestions I have gathered regarding events:

Adjust stockpile amount:
-offer the possibility to determine a granularity of the adjustment to be distributed to each storage in a loop. explanation: event distributes 100 wheat. the wheat will be sent to the lowest building ID storage. Allow the event to determines how much wheat should be distributed to ID1, then ID2, then ID 3.... Then ID 1 again. the amount is between 1 and 32. This allows to distribute evenly or not the adjustment of stockpile throughout the city.

Adjust desirability
-create effects to adjust the desirability value, range and step of a building type, or offset tile.

Allow an event to specifically check or target upgraded schools, upgraded markets, etc.

Trade

-Allow negative price for commodities
-Allow a trade city to return to future trade city
-Allow a trade route to close and reset the route opening fee
-Allow a levy on a given trade route ; allow the player to shut down trade on a given route

Editor UI: create a UI so the player may create and update 
-custom messages directly from the editor
-add a map description in the custom campaign kit


Time conditions:
-Allow an event to start on a given date (day/month/year)

**Reply by Johannes-callidus** [reply]
About time onditions thats already implemented with the "Time Passed" condition. And for adjust desirability, I opened a PR about that aswell as adjust cost and labor of a building type.

**Reply by Filu2259** [reply]
The time passed condition determines a number of months since the beginning of the game, right? 
But then, when you set days of trade interruption, its in days. but how many days in a month?
As is, it's a bit confusing without exterior info.

Also, is there a simple way to check for time passed after a trigger other than beginning of the game? Right now I have been using a custom value that adjusts monthly, but it boils down to setting up several events to engineer that condition.

**Reply by Johannes-callidus** [reply]
I'm not completly sure but I think one month has still 16 days.

**Comment by Miguelinileugim** [comment]
Your stone gate is neat but it looks like a cyclops loool

**Comment by Miguelinileugim** [comment]
> Depots indestructible and non-flammable (fire_proof = 1)

Trivial to implement, but why?

>Trade ships leave the city through the river exit point

This would require every scenario having several tiles marked as river entry and river exit then some tweak for routing. Not super hard, do you think it would be that much of a difference? If so, and you find me someone willing to go through each original scenario and add those extra points (we'll discuss it beforehand to figure out the format) then it should be doable yeah.

@eugenikus8

**Comment by Filu2259** [comment]
isn't it already the case that every map with sea trade requires river entry and river exit points?

**Comment by Filu2259** [comment]
action:  adjust city wages

**Comment by Filu2259** [comment]
action: adjust population sentiment: allow the event to adjust target sentiment score instead of actual sentiment score.

edit: 

Action: adjust target score instead of actual score for:  sentiment, health, prosperity, culture, other ratings with similar target systems.  a duration parameter would simplify the issue of synchronizing the event with the different update frequencies of the various ratings adjusted/ changed/set to value/not set to value

**Comment by Filu2259** [comment]
Would it be possible to place bridges in the editor (submenu with roads)? what about other buildings?

**Comment by PereDuchene** [comment]
Hey, new here but long time Caeser 3/Augustus player. Not sure if you're looking for suggestions here or if the Discord would be more appropriate but i've always felt one of the elements of Caeser 3 that is lacking is in terms of politics and the senate.

I wondered if it would be feasable and/or rewarding to do something similiar with the Senate building as we now have with buildings such as the Grand Temple/Pantheon/Colosseum and have certain 'perks' that can be invested in once it is finally built? Something along the lines of;

* Support for the Optimates - Your senate is stacked with blue-blooded conservatives who support the status quo; increased sentiment/housing level for patricians
* Support for the Populares - Your senate is stacked with forward-thinking reformers and rabble-rousers who appeal to the lower classes;  increased sentiment/housing level for citizens
* Appeal to the Equites - Your senate appeals to new money and dabbles in the grubby world of commerce more than is usually considered seemly; increased land/water trade

I wondered too if it might be worth splitting the senate into two buildings? An early game _Curio_ building that functioned like the senate does in game now and a later game 'Provincial Senate' building that is constructed in the same way as is a monument. This would reflect the fact that senates were not common throughout the Roman empire; or at least that is my understanding of them.

Onto a related issue I feel like the 'favour' mechanism is a bit outdated. It seems to essentially involve spamming the Emperor with gifts. I would potentially keep the gift mechanism but also put in place some ability, if feasible, to pay a tax or tribute to Rome on a yearly basis as being a way of increasing or decreasing favour. Potentially tied into the 'Curio' building.

This would reflect the development of the city from being;

Stage 1 - A frontier settlement and so not worth bothering with on a larger scale (i.e before the construction of a Curio)
Stage 2 - A growing city that is worth attention at an imperial level (stage reached when Curio is constructed and so taxes are paid to Rome)
Stage 3 - A true Roman city with its internal politics and own interests (stage reached once the 'Provincial Senate' is constructed)

Anyway this is the idea that has been chewing on my brain since I first played Augustus; hope you don't mind the suggestion!

**Reply by Johannes-callidus** [reply]
We are currently planning an expansion on the governance mechanics including changes in this field. I think that you are mentioning some good ideas here. We'll see what we can make out of this and more and take these into consideration.
Thank you.

**Reply by PereDuchene** [reply]
Thanks for reply! Can't wait to see what you're all cooking up :)
